### PR TITLE
Move spatial bounds and geometry calculations to :core:domain and remove GmsExt

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Towards https://github.com/google/ground-android/issues/3633

### Description
Removes presentation-layer `GmsExt.kt` and moves spherical geometry and bounding box calculations directly into `:core:domain`. This eliminates a Clean Architecture layer inversion where repositories and ViewModels depended on Google Maps SDK classes, while significantly reducing memory churn during map interactions.
### Performance Improvements
- **Zero-Allocation Map Render Loop (60fps)**: Pattern-matched `Bounds.contains(Geometry)` to avoid calling `getShellCoordinates()`. This eliminates ephemeral `listOf(coordinates)` allocations for `Point` LOIs and eager `ArrayList` flattening for `MultiPolygon` on camera pan/zoom updates in `BaseMapViewModel`.
- **$O(N) \to O(1)$ Heap Allocations in Spatial Math**: Eliminated transient Google Maps `LatLng` object allocations per vertex by computing areas and bounding boxes directly on domain primitives.
### Functional & Architectural Changes
- **Eliminated `GmsExt.kt`**: Migrated `area()`, `center`, and `contains()` to pure KMP domain models (`Geometry.kt`, `Bounds.kt`).
- **Spherical Area Parity**: Added `PolygonUtil.calculateSphericalPolygonArea` matching `SphericalUtil.computeArea` using Girard's theorem / polar triangle excess.
- **Bounds Factories & Anti-Meridian Handling**:
  - Replaced extension functions with explicit companion factories: `Bounds.fromCoordinates`, `fromGeometry`, and `fromGeometries`.
  - Added robust 180th meridian wrapping and normalization for `Bounds.fromCoordinates()`, `shrink()`, and `center`.
- **Testing**: Added comprehensive unit tests in `BoundsTest`, `GeometryTest`, and `PolygonUtilTest` covering all edge cases.

@andreia-ferreira  PTAL?

### CODE REVIEW

File: core/domain/src/commonMain/kotlin/org/groundplatform/domain/util/PolygonUtil.kt [hunk 1]
```diff
@@ -17,9 +17,48 @@ package org.groundplatform.domain.util
...
+  val num = t * sin(deltaLng)
+  val denom = 1.0 + t * cos(deltaLng)
+  return if (abs(num) < 1e-15 && abs(denom) < 1e-15) 0.0 else 2.0 * atan2(num, denom)
+}
+
+/**
+ * Returns the area of a closed path on Earth's surface in square meters, assuming a spherical
+ * Earth.
+ */
+fun calculateSphericalPolygonArea(
```
Comment (andreia-ferreira) [inline core/domain/src/commonMain/kotlin/org/groundplatform/domain/util/PolygonUtil.kt:42]: there are 2 different formulas to calculate areas in this file, this and `calculateShoelacePolygonArea`. Are they both needed or maybe we can opt to keep only one of them for all area calculations to avoid confusion?

Comment (shobhitagarwal1612) [reply core/domain/src/commonMain/kotlin/org/groundplatform/domain/util/PolygonUtil.kt:42]: They are not both needed, and we can definitely consolidate to `calculateSphericalPolygonArea`.

**Review by andreia-ferreira — APPROVED** [review]
awesome! 🚀
